### ISSUE TITLE
[Beta] Translating New React Docs

### ISSUE THREAD

**Issue by gaearon** [reactjs/react.dev#4135]
It's time to translate new React docs at https://react.dev!

* Translation status: https://translations.react.dev/
* How to start a new translation: https://github.com/reactjs/translations.react.dev

## Who is this issue for?

This issue is for existing translation repo maintainers.

## What is ok to translate now?

All pages in `src/content` are good to translate.

## Something unclear?

Please comment here with concerns or suggestions. Originally we wanted to suggest translating when more of the content is written. However, some translators have expressed interest in starting earlier so that's why we made this issue.

**Comment by arshadkazmi42** [comment]
Hi @gaearon 

Does that mean, we should stop the translation of the current website, which in progress? And start focusing on this?

**Comment by gaearon** [comment]
I think that would make the most sense. (Sorry if you spent a bunch of effort on this.)

**Comment by arshadkazmi42** [comment]
> I think that would make the most sense. (Sorry if you spent a bunch of effort on this.)

Thank you for confirmation. Will start working towards that.

**Comment by jakubdrozdek** [comment]
@gaearon Hey, I just started working on the Polish docs, but unfortunately my first PR's CI checks fail on `Analyze bundle` step.
https://github.com/reactjs/pl.reactjs.org/runs/4360102632?check_suite_focus=true
Do you know what's the problem?

**Comment by smikitky** [comment]
@jakubdrozdek That happened on our fork (Japanese), too. That [dawidd6/action-download-artifact@v2](https://github.com/dawidd6/action-download-artifact) GitHub action tries to download files (artifacts) that had been generated by Actions executed on the `main` branch. You got that error because no relevant workflow has been previously merged into the `main`. If I understand correctly, you can ignore that error and just merge for now. Everything should be fine from the next time.

**Comment by gaearon** [comment]
>Do you know what's the problem? Is it required to pass?

Like @smikitky said, try without it passing and see if that resolves next time. I wouldn't worry about it for now.

>And second question: where can we preview our translated website and what's the URL for the live Polish beta? 

If you write a few pages I'll set up a project. Unfortunately it's manual so write here and we'll do it.

**Comment by smikitky** [comment]
The current pre-commit hook (`/beta/.husky/pre-commit`) has the following two lines:

```
yarn generate-ids
git add -u src/pages/**/*.md
```

This may make sense on the main repo but makes no sense on translation forks. Every time I commit, I see many useless notices saying the explicit IDs are different from auto-generated ones.

https://github.com/reactjs/reactjs.org/blob/4022f9774b8dd93d863fd6bca6447b1f03ae3979/beta/scripts/generateHeadingIDs.js#L55-L57

What's worse, because of the unconditional `git add -u`, I accidentally committed a draft translation without knowing it. I think this process could be smarter.

**Comment by gaearon** [comment]
@smikitky yeah this sounds confusing! do you have a fix in mind? how did this work before?

**Comment by hg-pyun** [comment]
@gaearon Is there a schedule for the official release of new documents? It would be nice to share roughly.

**Comment by harish-sethuraman** [comment]
> When it finds a heading without an explicit ID, it should reject the commit instead of unconditionally fixing it and committing it. 

We might have to write a custom hook for that will look into that. 

>But in the meantime, I think this pre-commit hook could be simply removed

Will remove it

**Comment by jakubdrozdek** [comment]
@gaearon Hey, I just finished translating the whole first chapter. Do you think it's enough to deploy the Polish beta docs now? 😉 
https://github.com/reactjs/pl.reactjs.org/issues/287

**Comment by smikitky** [comment]
@gaearon Please set up the Japanese beta docs, too. Translations for several pages and MDX tags were merged into main.

**Comment by gaearon** [comment]
I actually just went on a vacation but I asked @rickhanlonii to find someone from the team to help with this. Ping me in twitter DM if there’s no progress in a week and I’ll double check.

**Comment by yepes** [comment]
I'd like to help with the Spanish translation. How can I get enrolled?

**Comment by rickhanlonii** [comment]
Just a heads up: we're going to be a bit delayed during React Conf this week. Thanks for your patience!

**Comment by dev-tom-0108** [comment]
Thanks so much

**Comment by rickhanlonii** [comment]
Sorry for the delay, Polish and Japanese docs are deployed here:
- https://pl-beta-reactjs-org.vercel.app/
- https://ja-beta-reactjs-org.vercel.app/

We're working to get the domains registered as well, will update when that's available.

**Comment by jakubdrozdek** [comment]
> Sorry for the delay, Polish and Japanese docs are deployed here:
> 
>     * https://pl-beta-reactjs-org.vercel.app/
> 
>     * https://ja-beta-reactjs-org.vercel.app/
> 
> 
> We're working to get the domains registered as well, will update when that's available.

Awesome! Thank you, that helps a lot :)

**Comment by rickhanlonii** [comment]
Heads up, the domains are now available:
- https://beta.pl.reactjs.org
- https://beta.ja.reactjs.org/

**Comment by smikitky** [comment]
@gaearon `Analyze bundle` job is still failing for every new PR made in the fork. Looks like the cause is this (Line 57):

https://github.com/reactjs/reactjs.org/blob/c005fa4d7a2a0a602d6d538526699d7073ae3109/.github/workflows/analyze.yml#L53-L60

Line 57 refers to `bundle_analysis_upload.yml`, but this workflow file no longer exists (removed by 25ac9d0). Perhaps the CI is currently using some stale artifact that remains only in the main repository as the "base".

**Comment by smikitky** [comment]
I made a PR to fix the problem above: #4192

**Comment by arshadkazmi42** [comment]
hi @gaearon 

For Hindi translation we have completed two pages, is it enough to get the hindi docs domain also setup?

https://github.com/reactjs/hi.reactjs.org/issues/168

**Comment by habarahonaa** [comment]
> I'd like to help with the Spanish translation. How can I get enrolled?

I second this! Would love to be part of it though I'm not currently a translation repo maintainer. Have a nice day everyone and thanks :)

**Comment by gaearon** [comment]
I've added more pages to the list of things that are good to start translating. See updated list above.

**Comment by rickhanlonii** [comment]
Hey all, sorry for the delay! These are now up:
- [beta.hi.reactjs.org](https://beta.hi.reactjs.org/learn/add-react-to-a-website)
- [beta.zh-hant.reactjs.org](https://beta.zh-hant.reactjs.org/learn/start-a-new-react-project)

**Comment by arshadkazmi42** [comment]
> Hey all, sorry for the delay! These are now up:
> 
> * [beta.hi.reactjs.org](https://beta.hi.reactjs.org/learn/add-react-to-a-website)
> * [beta.zh-hant.reactjs.org](https://beta.zh-hant.reactjs.org/learn/start-a-new-react-project)

Thank you for setting this up @rickhanlonii :smile:

**Comment by arshadkazmi42** [comment]
@gaearon @rickhanlonii 

Many test runs are failing for hindi translation repository on all PRs.
Is there anything needs to be done there?

Here is a sample PR https://github.com/reactjs/hi.reactjs.org/pull/186#pullrequestreview-886811263

**Comment by gaearon** [comment]
@arshadkazmi42 Please don't merge code changes like https://github.com/reactjs/hi.reactjs.org/pull/186. Only do translation changes on your repo. All code changes should be done on the main repo. Otherwise you'll get out of sync.

**Comment by arshadkazmi42** [comment]
@gaearon Got it. will take care of that. Is this why the builds are failing? I have been merging these Bump PRs.

**Comment by gaearon** [comment]
Depends on which failure specifically. You can click on each to see the cause.

Here is one, for example:

<img width="949" alt="Screenshot 2022-02-18 at 18 16 06" src="https://user-images.githubusercontent.com/810438/154739656-407a71c1-2d34-472a-896a-6533858037e7.png">

I would recommend to copy all content files outside the project, replace all the code with the original code in this repo, and then copy all the content files back. This way you'll get back to the state this repo is in.

**Comment by arshadkazmi42** [comment]
When I am trying to open those build, it does not shows me these details, seems there is some access level and I don't have access.
Below are the tests which are failing and circle ci is stuck
![Screenshot from 2022-02-18 23-48-17](https://user-images.githubusercontent.com/4654382/154740136-faefb8ba-6089-4524-8775-a22ce504eb33.png)
![Screenshot from 2022-02-18 23-48-25](https://user-images.githubusercontent.com/4654382/154740151-574b42d6-0329-45e4-9ca5-5872fe2a9aee.png)

**Comment by arshadkazmi42** [comment]
Also, when I am trying to access the details, for CircleCI and Vercel failed jobs, I get below pages, which does not shows any reason of failing
![Screenshot from 2022-02-18 23-50-37](https://user-images.githubusercontent.com/4654382/154740459-618b28ed-0d10-4bac-be42-9f7be258cd09.png)
![Screenshot from 2022-02-18 23-50-44](https://user-images.githubusercontent.com/4654382/154740467-14d1b7c1-3d92-4dd0-b81c-34539c2ae814.png)

**Comment by arshadkazmi42** [comment]
> Depends on which failure specifically. You can click on each to see the cause.
> 
> Here is one, for example:
> 
> <img alt="Screenshot 2022-02-18 at 18 16 06" width="949" src="https://user-images.githubusercontent.com/810438/154739656-407a71c1-2d34-472a-896a-6533858037e7.png">
> 
> I would recommend to copy all content files outside the project, replace all the code with the original code in this repo, and then copy all the content files back. This way you'll get back to the state this repo is in.

@gaearon 
I just did what you have suggested and created a PR. Still seeing some builds fail. Can you check the logs, what could be failing?

**Comment by arshadkazmi42** [comment]
Finally was able to fix the build issues with manual merge https://github.com/reactjs/hi.reactjs.org/pull/188. Thank you for the help

**Comment by carburo** [comment]
Hi @rickhanlonii, @gaearon! The Spanish translation has 3 finished articles. Could you set up deployments for the beta site?

**Comment by mirorauhala** [comment]
Hey @rickhanlonii, @gaearon! The [Finnish translation](https://github.com/mirorauhala/fi.reactjs.org/issues/1) has 5 pages currently translated. However, the pull request for it has not been approved yet. https://github.com/reactjs/reactjs.org-translation/pull/191

Perhaps you could look into it? Thanks!

**Comment by gaearon** [comment]
Sure, a PR is welcome.

**Comment by gaearon** [comment]
@mirorauhala Approved!

**Comment by theringsofsaturn** [comment]
@rickhanlonii I have finished 5 pages for the Albanian Beta website. I attached the PDF with the screenshots. It would be great if you please help us set up the deployment for this. @gaearon Even thought you added me to the maintainers list, I need someone to review and accept my pull requests, because I cannot. Could you please, take a look and accept it, when you have some time? It would be great if I had the rights to review and accept myself. Thank you in advance! 
[React Docs SQ.pdf](https://github.com/reactjs/reactjs.org/files/8890695/React.Docs.SQ.pdf)
Here are my commits:
https://github.com/reactjs/sq.reactjs.org/pull/43
[React Docs SQ.pdf](https://github.com/reactjs/reactjs.org/files/8890701/React.Docs.SQ.pdf)

**Comment by gaearon** [comment]
@theringsofsaturn Are you in the maintainers list? I think maintainers should be able to merge things. If you're not, can you send a PR to add yourself to the maintainers?

**Comment by EsmaeilSaleh** [comment]
Hello, I'm willing to translate to Persian/Farsi. But unfortunately the [current Persian repo](https://github.com/EsmaeilSaleh/fa.reactjs.org) doesn't even have the beta folder.
In recent weeks I've translated several pages (old docs) but they have never been reviewed.

Now I want to proceed with Beta version. Please add me to maintainers list.

**Comment by galnir** [comment]
Hey @gaearon, we have 3 translated pages in the Hebrew docs - https://github.com/reactjs/he.reactjs.org
Can we get a deployment?

**Comment by TajAlasfiyaa** [comment]
Hello Dan 
I want to translate to arabic but I have question is there RTL website version ? 
thank 
تاج الاصفياء اسحاق

**Comment by hg-pyun** [comment]
@gaearon 
Could you also set up the distribution of Korean beta documents?

https://github.com/reactjs/ko.reactjs.org

**Comment by faisalhani** [comment]
Hello @gaearon ,
The https://github.com/reactjs/ar.reactjs.org does not have the beta folder yet, I can help to contribute and help translate it into Arabic.

**Comment by awxiaoxian2020** [comment]
Hi, @gaearon The [zh-hans repo vercel](https://github.com/reactjs/zh-hans.reactjs.org/deployments/activity_log?environment=Production+%E2%80%93+zh-hans.reactjs.org) failed. I have no access for checking it. Could you please have a look?

**Comment by smikitky** [comment]
The language switch link seems to have disappeared from the new official site, and I didn't even know the URL would change to `react.dev` until just now. From the recent comments on this page, it looks like most translators haven't even gotten the beta site deployment previews up and running.

If the React team is still interested in maintaining URLs for sites in other languages, I think it's time to let us know the transition plan.

- Will there even be a language switch button on the new site?
- Will we get new URLs like `ja.react.dev`?
- Is it safe to keep working on the `beta` directory?

Also, I took a look at the [source for the home page](https://github.com/reactjs/reactjs.org/blob/main/beta/src/components/Layout/HomeContent.js), and I have to say that it is too unfriendly as a base for translations. Is the home page ready to be translated, or should we wait for the original team to do something about this?

**Comment by awxiaoxian2020** [comment]
The [zh-hans repo](https://github.com/reactjs/zh-hans.reactjs.org) has synced the new website. @gaearon Could you please set up it's vercel for the new docs to access https://zh-hans.reactjs.org?

**Comment by hg-pyun** [comment]
@gaearon Korean documents are also in a difficult situation because the beta folder has disappeared. Resolving conflicts will cause legacy documents to disappear, so quick feedback on this part is needed.

**Comment by smikitky** [comment]
@gaearon I really hate to say this, but do you even remember that dozens of volunteers around the world spend their time keeping translation forks up to date with the original repo you're currently working on?

Last week, the Japanese fork was cleanly up to date with the original repo, and this week, we got [this giant PR](https://github.com/reactjs/ja.reactjs.org/pull/546) with hundreds of conflicts and &gt;1400 file changes. Well, although I don't know how many hours it will take, it might be possible to merge this. But... **what's the point of deleting the legacy docs when you haven't even set up a deployment for the new site for each language**? Are you forcing us to shut down all the existing translation sites when new translated sites are still nowhere to be found?

I know the legacy docs code has been moved [here](https://github.com/reactjs/legacy.reactjs.org), and this means you have broken all the existing translation workflows without notice, and we have lost (or will soon lose) the means to update the legacy site (which is still the *current* site in all other languages other than English). Was this intentional?

Also, the repo name has been renamed to `react.dev`. Has the translation bot code been updated properly? Will we still get translation PRs next week and beyond?

**Comment by gaearon** [comment]
@awxiaoxian2020

I've switched https://zh-hans.reactjs.org/ settings, so it should show the new site now.

**Comment by smikitky** [comment]
@gaearon Thank you. Your plan should be good in the long run, but the biggest problem in the short run is that there is nothing we translators can do right now.

- If I merge [this huge PR](https://github.com/reactjs/ja.reactjs.org/pull/546) somehow, it will basically break every current setup (we can no longer update the old docs, and we can no longer get previews of the new docs, including branch previews). Note that we don't have access to Vercel's control panel (nor should we).
- Of course, if we don't merge the PR, we won't be able to do translations of the new site in a reasonable way using Git. This PR contains many critical updates including #5501.

Therefore, everything is blocked until we at least have a new repository for the legacy site (`ja.regacy.reactjs.org`) and assurance that Vercel will be set up promptly for both repositories.

**Comment by jhonmike** [comment]
I can help in the `pt-br`, until I start to correct the synchronization [PR](https://github.com/reactjs/pt-br.reactjs.org/pull/651), but it is very long, and I will take some time, about the deployment process and DNS configs, can you send me some doc to know what I should do, it was not very clear to me

**Comment by TajAlasfiyaa** [comment]
![image](https://user-images.githubusercontent.com/87016999/227042278-086182f3-e4f8-421a-b296-785fc09f4906.png)

I want to translate the new docs to arabic what can I do now ?
@gaearon

**Comment by gaearon** [comment]
We’ll do a pass over all sites tomorrow. @smikitky What do you propose is the best action? I’m thinking of creating new legacy repos and domains for all translations that are reasonably up-to-date (maybe >= 50%?). Not sure if they need to carry original commit access or we just leave them frozen. Could also make them subdirectories of the single legacy repo which might be nicer to avoid proliferation of repos. But this would mean they’re frozen. I think freezing them is fine but curious if you disagree.

**Comment by gaearon** [comment]
As for merge conflicts etc I’ll try to resolve them myself across all beta sites. (Good luck to me :)

**Comment by smikitky** [comment]
@gaearon Here's my suggestion:

- If at all possible, let's switch (back) to a monorepo approach (single repo for both docs for each language). It's the best option from a manageability perspective. We don't lose previous commits, we don't have to create dozens of new repos just for legacy docs, and we don't have to worry about keeping two different repos up to date. If that's no longer possible, then create `\[lang\].legacy.reactjs.org' for all forks with a reasonable amount of translation, as you suggest. I'm fine with losing earlier commits.
- If the PR generation bot is still working on Heroku, it's time to migrate to GitHub Actions. (The original source is [now private for no good reason](https://github.com/reactjs/reactjs.org-translation/commit/c49ead673b99e8977d9a99b89b81b1f2b346b139).) Ideally, we can set up this process by simply creating `.github/workflows/weekly-merge.yml` with something like `uses: reactjs/actions-weekly-merge@v1`.
- For the new docs, if you can resolve the conflicts, that's ideal. If you can't, that means no one else can, so I'm fine with "resetting" as you suggest.

**Comment by awxiaoxian2020** [comment]
The `zh-hans.react.org` has resolved the conflicts and the main branch been up to date with this repo. Thanks for @gaearon set up the site! And we create a [branch](https://github.com/reactjs/zh-hans.reactjs.org/tree/legacy-docs) to store our legacy docs. But I have to say, the legacy docs for zh-hans had a deployment problem at vercel before the new docs released.

**Comment by TajAlasfiyaa** [comment]
i am traslating to arabic language I will be ready when you finish
https://ar-reactjs.netlify.app/

**Comment by hg-pyun** [comment]
@gaearon 

I think it would be better to proceed in the same way as the current official English document. 
- Archive existing translation documents to a legacy repository.
- Create a new {lang}.react.dev repository.

I hope this work will be done quickly. We are ready to start translating. It would be very helpful if you could share even a minimum of milestones on how it is going to go.

**Comment by smikitky** [comment]
@hg-pyun In English, `react.dev` is not a new repository but renamed `reactjs.org`. But archiving `{lang}.reactjs.org` and starting new forks for new sites seems like another good option to me.

So I'm seeing there are four suggested approaches so far:

1. Use monorepo, maintain two directories for two sites (my suggestion)
2. Maintain two branches in one repository, `main` and `legacy` (zh-hans did this already)
3. Keep existing `{lang}.reactjs.org` as-is (or archive/rename them), create new forks like `{lang}.react.dev` and start from scratch there (@hg-pyun's suggestion)
4. Rename `{lang}.reactjs.org` to `{lang}.react.dev`, merge conflicts. Make a new repository (or many repositories) for legacy docs and copy legacy contents there (@gaearon's suggestion)

My personal preference is 1 > 3 > 2 > 4.

| | 1 | 2 | 3 | 4 |
| - | - | - | - | - |
| history of legacy docs | preserved | preserved | preserved | lost |
| num. of language forks | preserved | preserved | doubles | doubles |
| future merging tasks | very simple | can be tricky | simple | simple |

At any rate, we need to decide quickly. As I explained earlier, the existing workflow is already almost broken, and the maintainers of each language independently seem to have started to resolve it according to what they think is best. I'm starting to see a chaotic situation. Until agreement is reached on a direction, I'll just stop all work and wait.

**Comment by jhonmike** [comment]
@gaearon 
I fixed the sync [PR](https://github.com/reactjs/pt-br.reactjs.org/pull/651) and [PR](https://github.com/reactjs/pt-br.reactjs.org/pull/652) conflicts and merged it, I also made a [legacy-docs](https://github.com/reactjs/pt-br.reactjs.org/tree/legacy-docs) branch, now I need help publishing the [site](https://pt-br.reactjs.org/), after the [site](https://pt-br.reactjs.org/) is published I will sync the translations, help me 🙏

**Comment by gaearon** [comment]
A few quick thoughts:

- Moving to another repo does not erase history. If we were to move legacy docs to other repos, we'd keep all existing commits there.
- However, I don't think we want to double the amount of repos just for the sake of legacy sites.
- I don't love the idea of having two directories in a monorepo because it gives the impression that both sites have the same importance and are actively maintained. I think we'd like the translation forks to 1:1 mirror the new site structure.
- However, we'd like existing translators to have full commit access to their legacy translations — in case they want to continue some work on it. For example, it would be nice to at least get to the point that all existing "go to new docs!" banners that I've added all over the legacy site to be there in the legacy translations too.
- We don't plan to be updating the original legacy site from this point on. Except super minor things. So it seems ok for it to be excluded from the syncing workflow. You're not gonna need to "catch up" with it anymore.

With these constraints, my favored option is moving the old site to a `legacy` branch for each translation repo. There will be no auto-syncing for it but it's ok because we won't update upstream either. It leaves the translators the option to keep working on it, but it won't be very noticeable to new contributors.

I'll try to get this setup fully working with the Chinese fork first. If it works well, I can do a pass to get all other forks to match the same setup.

**Comment by jhonmike** [comment]
Great @gaearon
But it would be possible to release the domain https://pt-br.reactjs.org or https://pt-br.react.dev for the [pt-br repo](https://github.com/reactjs/pt-br.reactjs.org)?

I ended up synchronizing with the main and I have the site down because of an error in vercel that I can't see due to lack of permission

**Comment by gaearon** [comment]
Yea we'll be doing a pass to fix them all.

**Comment by awxiaoxian2020** [comment]
> I'll try to get this setup fully working with the Chinese fork first. If it works well, I can do a pass to get all other forks to match the same setup.

Hi, @gaearon just let you know, we have made a [branch](https://github.com/reactjs/zh-hans.reactjs.org/tree/legacy-docs) for legacy docs in `zh-hans` repo.

**Comment by gaearon** [comment]
For some reason I can't get that branch to build, but not sure why. Any ideas? Are `package.json` and `yarn.lock` identical to legacy docs repo? Anything else I could be missing?

**Comment by awxiaoxian2020** [comment]
> For some reason I can't get that branch to build, but not sure why. Any ideas? Are `package.json` and `yarn.lock` identical to legacy docs repo? Anything else I could be missing?

Actually, I comment for this failure when we don't release the new site. Now I delete it for clarifying. Even I email you for this before.

I need some vercel error message to confirm what happened. Could you please share it? @gaearon

**Comment by awxiaoxian2020** [comment]
Good! the legacy docs for zh-hans was fixed https://github.com/reactjs/zh-hans.reactjs.org/commit/0222f7cb216dd5b008b93eb7fa313c7f080bd68c by @rickhanlonii . Thanks so much indeed!

But I think we should just run the vercel https://zh-hans.legacy.reactjs.org/ on the legacy branch not on main branch. Could you please set up it?

**Comment by gaearon** [comment]
Sorry I'm a bit confused by the current setup. We have:

- https://zh-hans.legacy.reactjs.org/ running from the `legacy` branch
- https://zh-hans.reactjs.org/ running from the `main` branch

I also see another Vercel project in commit CI deployments (https://vercel.com/docschina/zh-hans-reactjs-org) but I don't have access to it (it doesn't belong to Meta org). Not sure how this part works.

What else is needed to do on our side?

**Comment by awxiaoxian2020** [comment]
> Sorry I'm a bit confused by the current setup. We have:
> 
> * https://zh-hans.legacy.reactjs.org/ running from the `legacy` branch
> * https://zh-hans.reactjs.org/ running from the `main` branch
> 
> I also see another Vercel project in commit CI deployments (https://vercel.com/docschina/zh-hans-reactjs-org) but I don't have access to it (it doesn't belong to Meta org). Not sure how this part works.
> 
> What else is needed to do on our side?

You should remove https://zh-hans.legacy.reactjs.org/ from main branch and remove https://zh-hans.reactjs.org/ from legacy branch to avoid wrong ci messages.

The docs-china vercel is a temp deployment by ourselves in previous time. You don't need to care about it. We will fix or remove it. Thanks very much indeed!

**Comment by gaearon** [comment]
>You should remove https://zh-hans.legacy.reactjs.org/ from main branch and remove https://zh-hans.reactjs.org/ from legacy branch to avoid wrong ci messages.

Ah. I'm not sure if this is possible. Do you know how to do it?

**Comment by awxiaoxian2020** [comment]
> > You should remove https://zh-hans.legacy.reactjs.org/ from main branch and remove https://zh-hans.reactjs.org/ from legacy branch to avoid wrong ci messages.
> 
> Ah. I'm not sure if this is possible. Do you know how to do it?

Oh! I'm not familiar with vercel. I will look into it for a better workaround.

**Comment by gaearon** [comment]
Maybe something like workarounds at the end of https://github.com/vercel/vercel/issues/3166 can work. It seems like we'd like to the new site to ignore the `legacy` branch, and for the legacy site to ignore all non-`legacy` branches. Is that a good summary of what we want?

**Comment by awxiaoxian2020** [comment]
> Maybe something like workarounds at the end of [vercel/vercel#3166](https://github.com/vercel/vercel/issues/3166) can work. It seems like we'd like to the new site's CI to ignore the `legacy` branch, and for the legacy site's CI to ignore all non-`legacy` branches. Is that a good summary of what we want?

Yes, you are right!

**Comment by awxiaoxian2020** [comment]
> This might still make PRs to legacy awkward but that seems like a corner case. Unless we come up with some way to distinguish them other than by branch name — e.g. by dir structure.

From our side (zh-hans community), we don't accept PRs for legacy branch often. So it's a corner case indeed for zh-hans repo.

**Comment by gaearon** [comment]
OK I think I got it working. For future reference, I put this into Vercel's "Ignored Build Step" setting:

* For legacy site: `bash -c 'if [[ -d "gatsby" ]]; then exit 1; else exit 0; fi'` ("build only if `gatsby` dir exists")
* For new site: `bash -c 'if [[ -d "gatsby" ]]; then exit 0; else exit 1; fi'` ("build only if `gatsby` dir doesn't exist")

This should ensure it works for PRs to either branch too — by checking if `gatsby` dir exists (which we won't remove on the legacy site). With this approach, the checks still appear in CI but they are immediately green with "Canceled by Ignored Build Step". Let me know if this works on your side — I triggered a few commits and it seems fine to me.

**Comment by awxiaoxian2020** [comment]
> OK I think I got it working. For future reference, I put this into Vercel's "Ignored Build Step" setting:
> 
> * For legacy site: `bash -c 'if [[ -d "gatsby" ]]; then exit 1; else exit 0; fi'` ("build only if `gatsby` dir exists")
> * For new site: `bash -c 'if [[ -d "gatsby" ]]; then exit 0; else exit 1; fi'` ("build only if `gatsby` dir doesn't exist")
> 
> This should ensure it works for PRs to either branch too — by checking if `gatsby` dir exists (which we won't remove on the legacy site). With this approach, the checks still appear in CI but they are immediately green with "Canceled by Ignored Build Step". Let me know if this works on your side — I triggered a few commits and it seems fine to me.

It works fine! Thanks!

**Comment by gaearon** [comment]
Ok cool. Seems like we got one setup working. I'll try to replicate it for others tomorrow. (If someone has objections, lmk!)

**Comment by awxiaoxian2020** [comment]
@gaearon I found [the PR to main branch](https://github.com/reactjs/zh-hans.reactjs.org/pull/1101) still didn't ignore legacy deployment.

**Comment by smikitky** [comment]
What's the URL for the new site? I assumed it would be `{lang}.react.dev`, but are you replacing the contents of `{lang}.reactjs.org` with new contents tomorrow, for all languages?

It's fine to unconditionally set up new `{lang}.react.dev` deployments tomorrow, but it's too early to unconditionally replace `{lang}.reactjs.org` for all sites. Zh-hans seems to be exceptionally well translated already (congratulations!), but I think this is not the case with many other languages. The Japanese docs still need a month or so for translation.

**Comment by smikitky** [comment]
I think a safer approach would be like this:

- Unconditionally set up a new deployment `{lang}.reactjs.dev` from the `main` branch (whether or not the new content has been already merged into it)
- Create **two** branches, `legacy` (if it doesn't exist yet) and `reactjs.org`
  - The `legacy` branch is deployed to the new URL, `{lang}.legacy.reactjs.org`
  - The `reactjs.org` branch is deployed to the existing URL, `{lang}.reactjs.org`. This URL will eventually be redirected to `{lang}.reactjs.dev`, but the timing is left to the discretion of each language's maintainers.

**Comment by gaearon** [comment]
>I found https://github.com/reactjs/zh-hans.reactjs.org/pull/1101 still didn't ignore legacy deployment when the contributor has no write access.

This looks to be due to something touching `package.json` (maybe one of the merge commits?) In that case, Vercel asks for deployment approval before checking whether CI should run at all.

One thing we could do is turn off these deployment approvals altogether. They don't make sense (IMO) because we don't use any environment variable secrets.

**Comment by rickhanlonii** [comment]
@smikitky I think we just need two branches:

- `main`: deployed to `{lang}.react.dev`
- `legacy`: deployed to `{lang}.legacy.reactjs.org`

Then, in the domain config, we add the following redirect rule:
  - `{lang}.reactjs.org/*` -> `{lang}.legacy.reactjs.org` 

Finally, once the new docs are done, we add an additional rule for the homepage:
  - `{lang}.reactjs.org` -> `{lang}.react.dev` (homepage)

If the maintainer wants, they can also setup all the other redirect rules [here](https://github.com/reactjs/legacy.reactjs.org/blob/main/vercel.json).

Would that work or am I missing something?

**Comment by arshadkazmi42** [comment]
@gaearon 

I am maintaining Hindi translations and I do not see any sync PR for the above discussion.

Do I need to do anything at my end for the above discussion?

**Comment by gaearon** [comment]
I'll keep working on this throughout the week — I'll let you know the progress as we look at each repo.

**Comment by smikitky** [comment]
@rickhanlonii Who will change this redirect rule when the new docs are done? We do not have access to the Vercel dashboard, including the domain config. It's not very ideal for the maintainers of each language to ask gaearon or someone else to make this change and wait. The third branch will allow us to change the redirect rules using [this feature](https://vercel.com/docs/concepts/projects/project-configuration#redirects).

**Comment by gaearon** [comment]
>Besides, your approach means that people will immediately start to see "legacy" in the URL, even though there are no newer documents yet. That's confusing. The most recent documents should not be called "legacy".

I think we'd like to make an editorial decision here that we _should_ be showing "legacy" because we need to communicate the docs are out of date compared to how we suggest using React. Yes, there may be no newer docs yet, but then people are at least informed and have the option of reaching for automatic translations or trying to read the original docs.

**Comment by smikitky** [comment]
@gaearon I doubt that alone justifies changing the URL now and changing it again in the near future. For that purpuse, the legacy docs already have a banner and "Notes" in almost all pages.

**Comment by smikitky** [comment]
That being said, the URL change is a minor problem, and you can proceed as you wish. As long as the current translated docs won't suddenly disappear today, we can manage. The third branch is not necessary if someone can quickly modify the redirect rule upon request when the new docs are ready.

**Comment by gaearon** [comment]
Hi folks. My apologies for the delay. We’ve run into some issues with our domain configuration and we’re waiting for that to get resolved. We expect it to be fixed next week.

**Comment by david-mbugua** [comment]
Hi @gaearon , how does someone become a translator? I have seen there's no Swahili translation yet and It's the most spoken language in Africa. I would like to contribute to that. If it's already in the works, I would appreciate it if you can lead me to the right repository and I can help contribute.
Thanks.

**Comment by awxiaoxian2020** [comment]
> Update the issue template to include all current pages, and sync it across all translation repos. Make sure all translation repos contributor lists are up to date.
@gaearon I know you could be busy, but I want to see it indeed.

**Comment by awxiaoxian2020** [comment]
Hi, folks! I make a translation list, here is the [raw](https://gist.githubusercontent.com/awxiaoxian2020/bc7bd9a3f8cf6f9f9456b2d83f7cd860/raw/9d1f328211b7e374acc28cd7ba10fb229b73322d/React%2520Translation%2520List). You can replace `react.dev` with your domains like `zh-hans.react.dev`. I hope this can help you!

**Comment by gaearon** [comment]
Thank you @awxiaoxian2020, that's very helpful! I'll try to go through the repos asap — just back from holidays.

**Comment by agmmnn** [comment]
> Hi, folks! I make a translation list, here is the [raw](https://gist.githubusercontent.com/awxiaoxian2020/bc7bd9a3f8cf6f9f9456b2d83f7cd860/raw/9d1f328211b7e374acc28cd7ba10fb229b73322d/React%2520Translation%2520List). You can replace `react.dev` with your domains like `zh-hans.react.dev`. I hope this can help you!

Also to be more specific, these can be add as subcategories to the index.

`TopNav.tsx`, `Footer.tsx`, `sidebarBlog.json`, `sidebarCommunity.json`, `sidebarHome.json`, `sidebarLearn.json`, `sidebarReference.json`

**Comment by gaearon** [comment]
We've started going through the repos. It's going to be a bit churn-ey as we resolve Git conflicts. Please be patient with us.

**Comment by emersonbroga** [comment]
Hi!!! 

I want to help with the Translation to pt-BR 🇧🇷, is there a translation guide on how to get started?

I saw this repo [zh-hans.react.dev](https://github.com/reactjs/zh-hans.react.dev) but I couldn't find a pt-br.react.dev repo that I could fork and create PRs. Whats the process I should follow? Please advise.

Note: [I've forked this repo and started translating two sidebars (not much, but its a start)](https://github.com/reactjs/react.dev/pull/5934/files). 

Also I just submitted the  Individual Contributor License Agreement.

**Comment by gaearon** [comment]
We've figured out the full process to get a single translation in sync with the upstream. We're going to do it manually for translations that are relatively complete (including legacy and modern ones). Most of the work is in resolving merge conflicts, which we're going to do ourselves. For example, see https://github.com/reactjs/es.react.dev/pull/654 and https://github.com/reactjs/es.react.dev/pull/655.

For future reference, here is the complete process we're doing:

<details>

* Legacy site
    * Create a legacy branch in GitHub repo
        * Should have old folder structure
        * Should not have a `beta` directory
        * Get it up to date (and resolve conflicts) so it has the red banners. Example: https://github.com/reactjs/es.react.dev/pull/654
        * Ensure `site-constants.js` has domain set to `lang.legacy.reactjs.org`
        * Ensure `vercel.json` matches the English site’s exactly
        * If modern site is >50% translated
            * Edit `vercel.json` to have appropriate redirects. Specifically:
                * Find all redirects with `reactjs.org` inside `has` → `host` conditions. For each:
                    * Replace `reactjs.org` with `lang.reactjs.org` in `host` entries
                    * Replace `react.dev` with  `lang.react.dev` in destination
                    * Replace `legacy.reactjs.org` with `lang.legacy.reactjs.org` in destination
    * Find Vercel project for legacy site
        * Set Vercel project name to `lang-legacy-reactjs-org`
        * General → Build & Dev Settings → Gatsby
        * Git → Production Branch → `legacy`
        * Git → Ignored Build Step → `bash -c 'if [[ -d "gatsby" ]]; then exit 1; else exit 0; fi'`
        * Security → Git Fork Protection → Off
        * Force redeploy, ensure CI builds
        * Domains + Redirect
            * Main domain: `lang.legacy.reactjs.org` (assigned to `legacy` branch)
            * Alternative domain: `lang.reactjs.org`
                * If new site is translated > 50%
                    * Assign `lang.reactjs.org` domain to `legacy` branch
                * If new site is translated < 50%
                    * Make `lang.reactjs.org` a 307 temporary redirect to `lang.legacy.reactjs.org`

* Modern site
    * Open the main branch in GitHub repo 
        * Is the modern site translated > 50%?
            * If yes
                * Add it to `deployedTranslations` in English translation’s `SEO.tsx` (and sync)
            * If no, and there has been zero translation effort at this point
                * Close all sync PRs because we’re about to reset
                * Reset `main` to the English upstream to kickstart translation
        * Regardless, get it up to date:
            * Ensure it has a new folder structure
            * Ensure `vercel.json` matches the English site’s exactly
            * Ensure `src/siteConfig.js` has correct `languageCode` and `hasLegacySite` settings
            * Get it up to date with design so it has homepage etc. Example: https://github.com/reactjs/es.react.dev/pull/655
            * Verify sidebar translations are still there
            * Update repo name to `lang.react.dev`
    * Find or create a Vercel project for the modern site
        * Set Vercel project name to `lang-react-dev`
        * General → Build & Dev Settings → Next.js
        * General → Root Directory → (empty)
        * Git → Production Branch → `main`
        * Git → Ignored Build Step → `bash -c 'if [[ -d "gatsby" ]]; then exit 0; else exit 1; fi'`
        * Security → Git Fork Protection → Off
        * Force redeploy, ensure CI builds
        * Domains
            * If the modern site is translated > 50%:
                * Add and set `lang.react.dev` as primary domain
                * If `beta.lang.reactjs.org` is there, make it 308 redirect to `lang.react.dev`. No need to add one if it’s not there.
            * Otherwise, no domains — just keep deployment previews

* Checklist
    * [lang.legacy.reactjs.org](http://lang.legacy.reactjs.org/) shows banners
    * [lang.reactjs.org](http://lang.reactjs.org/) is a redirect → either to [lang.react.dev](http://lang.react.dev/) (if >50%) or [lang.legacy.reactjs.org](http://lang.legacy.reactjs.org/)
    * [lang.reactjs.org/docs/state-and-lifecycle.html](http://lang.reactjs.org/docs/state-and-lifecycle.html) is a redirect → [lang.legacy.reactjs.org/docs/state-and-lifecycle.html](http://lang.reactjs.org/docs/state-and-lifecycle.html)
    * [lang.react.dev](http://lang.react.dev/) (if >50%) should show new homepage, otherwise should not exist
    * Should be able to navigate around both domains
    * Check `<meta rel="canonical">` on both
    * Add a “Test CI” commit to both legacy and main branch (e.g. adding a newline to README), verify they’re green, revert it
    * Write on the GH translation thread
        * Example comment: https://github.com/reactjs/es.react.dev/issues/500#issuecomment-1513674006

</details>

No further action is needed from the translators at this point. We'll try to get through them all in a few days.

Thanks everyone for being patient with us!

**Comment by gaearon** [comment]
@emersonbroga The `pt-br` repo is here: https://github.com/reactjs/pt-br.reactjs.org. Please coordinate with its maintainers. Thank you!

**Comment by alioguzhan** [comment]
Hi @gaearon, I merged those two PRs from @rickhanlonii (https://github.com/reactjs/tr.reactjs.org/pull/395 and https://github.com/reactjs/tr.reactjs.org/pull/396)

Now, I ended up with an up-to-date `legacy` branch and a **broken** https://tr.reactjs.org.

A little guidance would be helpful now. Since the website is completely down. 

Thanks

**Comment by gaearon** [comment]
@alioguzhan Sorry, we should have been clearer — we ask **not to merge anything because we're going to go through all of the catch-up ourselves.** (Like mentioned in this thread, no action is needed on the translators' part.) I'll try to see if I can get these fixed quickly but I can't promise — we might have to wait until tomorrow.

**Comment by gaearon** [comment]
@alioguzhan Should be working now.

**Comment by alioguzhan** [comment]
Yes, it is working. Thanks for helping! @gaearon

**Comment by gaearon** [comment]
Alright everyone, here's an update!

First — really appreciate y'all's effort on the old translations. We've tried to honor it the best we could by doing manual merges and trying to get every repo with > 50% translations up-to-date and running (both versions).

Each repo now has a `legacy` branch (effectively frozen) and a `main` branch (where we resolved all conflicts). The `legacy` branch is deployed as `lang.legacy.reactjs.org`, while the `main` branch is at `lang.react.dev`.

## Up-to-date repositories

We have finished getting these repos up-to-date:

- https://github.com/reactjs/zh-hans.reactjs.org
- https://github.com/reactjs/es.react.dev
- https://github.com/reactjs/tr.react.dev
- https://github.com/reactjs/pl.react.dev
- https://github.com/reactjs/it.react.dev
- https://github.com/reactjs/ja.react.dev
- https://github.com/reactjs/ar.react.dev
- https://github.com/reactjs/ko.react.dev
- https://github.com/reactjs/pt-br.react.dev
- https://github.com/reactjs/ru.react.dev
- https://github.com/reactjs/az.react.dev
- https://github.com/reactjs/fr.react.dev
- https://github.com/reactjs/hu.react.dev

This means **you are all set for translating new docs now**.

We are currently working to finish work on these repos — for now their deploys are broken:

- https://github.com/reactjs/mn.react.dev
- https://github.com/reactjs/uk.react.dev
- https://github.com/reactjs/zh-hant.react.dev
- https://github.com/reactjs/id.react.dev
- https://github.com/reactjs/vi.react.dev
- https://github.com/reactjs/de.react.dev
- https://github.com/reactjs/he.react.dev
- https://github.com/reactjs/fa.react.dev
- https://github.com/reactjs/hi.react.dev

We'll try to get them all fixed ASAP.

## Archived translations

Unfortunately, it is not feasible for us to do so much manual work for every single existing translation repo. We've decided to archive the rest of translations and unlink them from the domain to avoid misleadingly stale information from appearing official. We've archived all repositories that had < 50% translated content (except a couple of them which already started new site translations). These repositories are still available, so if you would like to pick them up (or if you were their original maintainer), please register your intent on https://github.com/reactjs/translations.react.dev. If these languages have a committed maintainer crew ready to take on translating the new site, we'd be happy to set them up again.

- https://github.com/stale-reactjs-org-translations/bn.reactjs.org
- https://github.com/stale-reactjs-org-translations/bg.reactjs.org
- https://github.com/stale-reactjs-org-translations/ca.reactjs.org
- https://github.com/stale-reactjs-org-translations/el.reactjs.org
- https://github.com/stale-reactjs-org-translations/km.reactjs.org
- https://github.com/stale-reactjs-org-translations/tl.reactjs.org
- https://github.com/stale-reactjs-org-translations/gu.reactjs.org
- https://github.com/stale-reactjs-org-translations/ht.reactjs.org
- https://github.com/stale-reactjs-org-translations/hy.reactjs.org
- https://github.com/stale-reactjs-org-translations/ka.reactjs.org
- https://github.com/stale-reactjs-org-translations/kn.reactjs.org
- https://github.com/stale-reactjs-org-translations/ku.reactjs.org
- https://github.com/stale-reactjs-org-translations/lt.reactjs.org
- https://github.com/stale-reactjs-org-translations/ml.reactjs.org
- https://github.com/stale-reactjs-org-translations/ne.reactjs.org
- https://github.com/stale-reactjs-org-translations/nl.reactjs.org
- https://github.com/stale-reactjs-org-translations/pt-pt.reactjs.org
- https://github.com/stale-reactjs-org-translations/ro.reactjs.org
- https://github.com/stale-reactjs-org-translations/si.reactjs.org
- https://github.com/stale-reactjs-org-translations/sv.reactjs.org
- https://github.com/stale-reactjs-org-translations/ta.reactjs.org
- https://github.com/stale-reactjs-org-translations/te.reactjs.org
- https://github.com/stale-reactjs-org-translations/th.reactjs.org
- https://github.com/stale-reactjs-org-translations/ur.reactjs.org
- https://github.com/stale-reactjs-org-translations/uz.reactjs.org
- https://github.com/stale-reactjs-org-translations/mr.reactjs.org
- https://github.com/stale-reactjs-org-translations/my.reactjs.org
- https://github.com/stale-reactjs-org-translations/sq.reactjs.org
- https://github.com/stale-reactjs-org-translations/sr.reactjs.org
- https://github.com/stale-reactjs-org-translations/sw.reactjs.org
- https://github.com/stale-reactjs-org-translations/tg.reactjs.org

## Workflows

There's a bunch of TODOs we need to do after. I've documented them in https://github.com/reactjs/translations.react.dev/issues/256.

**Comment by TheBrown** [comment]
Hi @gaearon, 

I want to open a new branch for a new language, Lao (it's going to be lo.react.dev).
Could you please create a new branch for me? I want to contribute to translating a ReactJS website into Lao. I tried to create a new pull request on the legacy reactjs web but got rejected, and you suggested creating a new PR. I decided to go with the new reactjs website instead (I understand that the previous one is still beta, right?)

I hope this will be a chance to help grow the ReactJS community in Laos.

My translation experience, ref: [contributors](https://pontoon.mozilla.org/lo/contributors/)

**Comment by gaearon** [comment]
Please wait, we'll get them merged ourselves.

**Comment by gaearon** [comment]
We've updated `main` and `legacy` branches for all active languages and closed all stale sync PRs. As the next step, we will update the issue templates across repos to be consistent and reflect the current set of pages on the website. Then we'll fix https://translations.react.dev to show the correct percentages, and ensure that the repo permissions set from https://github.com/reactjs/translations.react.dev are actively synced.

**Comment by gaearon** [comment]
We've updated the translation checklist issue across all repositories, fixed some lost translated pages, and fixed the global progress tracker at https://translations.react.dev/ to show the status of new translations!

<img width="1432" alt="Screenshot 2023-04-25 at 05 37 02" src="https://user-images.githubusercontent.com/810438/234175735-00a8c11f-4273-4741-bd13-de862c43de93.png">

**Comment by tdd** [comment]
Is there any way to enable some smart quoting for Markdown contents, as was the case for legacy docs? I'm looking in particular for apostrophes in text becoming `’` instead of of ASCII single quote (`'`).

We try to honor French typographic code to the best of our abilities in our translations, the legacy docs had it, but for now we have to manually type the proper apostrophe or search-and-replace it before pushing or at PR review time.

/poke @gaearon

**Comment by gaearon** [comment]
We've fixed the infrastructure for creating new translations: https://github.com/reactjs/translations.react.dev/pull/285. We've also moved content sync scripts to run on a public repo so that it's more noticeable if they start failing again: https://github.com/reactjs/translations.react.dev/actions. We're no longer keeping the maintainer lists in a centralized repo — they can now be managed by individual teams using each project's Settings, which is how people have already been de facto updating them. (I'll remove the separate concept of "teams" later but for now it doesn't hurt.)

If you want to start a new translation, please consider looking through https://github.com/reactjs/translations.react.dev/pulls for calls for volunteers, or submit your own request (ideally you'd have some translation experience before volunteering as a maintainer).

**Comment by gaearon** [comment]
@tdd I think smart quoting works for English, can you check that? If it doesn't work for French, maybe we can find where it's happening (I suspect some MDX plugin) and adjust the logic?

**Comment by faisalhani** [comment]
hello @gaearon I would like to contribute to the translation on Arabic Lang, please add me to the maintainers group if available.

**Comment by gaearon** [comment]
@faisalhani Please discuss this in the Arabic language repository. Existing maintainers make a decision on whether to add more maintainers.

**Comment by Kamilius** [comment]
@gaearon I would like to start translating into Ukrainian. Is there any place I to ask questions besides this thread? (Discord, etc.?).
I've already read through the guidelines, but still don't know how to let the other repo maintainers know, that some part of the website is already being worked on.

**Comment by gaearon** [comment]
@Kamilius The current Ukrainian repo maintainers are @berzavlu ldmrkl @ithompson4 @oleksii-polovyi @MargarytaChepiga. The issue you should be using is https://github.com/reactjs/uk.react.dev/issues/457. If they are not responsive in the issue for several weeks, please file a new issue in the repo to form a new team of maintainers. Once you have assembled a team of maintainers, please let us know by filing an issue in https://github.com/reactjs/translations.react.dev, and we'll give you the rights.

**Comment by smikitky** [comment]
Hi. I made this little tool that translates a Markdown file into another language using the ChatGPT API.

**ChatGPT Markdown Translator**: https://github.com/smikitky/markdown-gpt-translator

Of all the AI products I know, ChatGPT (especially GPT-4) provides the most accurate translations, but it unfortunately cannot process long texts in one go. This tool splits a Markdown file into segments, translates them in parallel, and combines the results.

I have tried it with a dozen files, and although it still makes mistakes, I found it can significantly reduce the time needed to finish the translation. If you are interested, please give it a try.

As an example, [this page](https://ja-react-8aa88t4yk-fbopensource.vercel.app/learn/tutorial-tic-tac-toe) was created by translating the original Tic-tac-toe Tutorial using this tool (with GPT-4) and spending a few minutes fixing format errors.

**Comment by gaearon** [comment]
Amazing, thank you for sharing! FWIW, my stance on this is that it doesn't matter how the translation gets generated as long as there are humans in the loop that ensure the terminology is consistent, the style is pleasant to read and idiomatic for the target language, and that inaccuracies don't creep in.

**Comment by yousufkalim** [comment]
Dear @gaearon,

I am writing to express my interest in volunteering to create/translate React documentation in Urdu language. I am a native Urdu speaker and have been using React for several years. I am confident that I can create clear and concise documentation that will be helpful to a lot of Urdu speakers who are learning and using React.

I’m the author of the most famous [react-easy-router](https://www.npmjs.com/package/react-easy-router) library with around 4 million downloads a week, [@yousufkalim](https://github.com/yousufkalim) is my GitHub profile, where you can see my other work and famous open-source libraries built for react.

I am available to start working on the documentation immediately. I am also willing to work on other projects related to React, such as creating tutorials, writing blog posts, or giving talks.

Thank you for your time and consideration. I look forward to hearing from you soon.

Sincerely,

Yousuf Kalim

**Comment by gaearon** [comment]
@yousufkalim To start a new translation, please see the process in https://github.com/reactjs/translations.react.dev.

**Comment by ysh199882** [comment]
> We've updated the translation checklist issue across all repositories, fixed some lost translated pages, and fixed the global progress tracker at https://translations.react.dev/ to show the status of new translations!我们更新了所有存储库的翻译清单问题，修复了一些丢失的翻译页面，并修复了 https://translations.react.dev/ 上的全局进度跟踪器以显示新翻译的状态！
> 
> <img alt="Screenshot 2023-04-25 at 05 37 02" width="1432" src="https://user-images.githubusercontent.com/810438/234175735-00a8c11f-4273-4741-bd13-de862c43de93.png">

Hello,

I am a loyal user of React. I would like to contribute to the translation of the React documentation. However, I noticed that there used to be a progress indicator for Simplified Chinese translation in this image, but now it's gone. Was there any particular reason for this change?

**Comment by smikitky** [comment]
Hi, I made this small tool to translate OGP images stored as `public/images/og-*.png`. They are used by Facebook and other SNS.

https://codesandbox.io/p/sandbox/8299gn

You can generate images like this in 30 seconds:

![image](https://github.com/reactjs/react.dev/assets/7779406/efa2bd10-178e-4072-bf30-4152a2c9b7b0)

RTL and composite fonts are not supported, you can edit the source code to suit your needs.

**Comment by smikitky** [comment]
By the way, if there are translators who are wondering why the new translations are not showing up on Google, please refer to #6260.

**Comment by tdd** [comment]
Thank you for this! Just used it for the FR translation 😉

**Comment by fabiarias** [comment]
I would like to help with the translate.

**Comment by MiryamMann** [comment]
Is there a translation in Hebrew already? 
I'll be happy to translate articles in React web to Hebrew.
 Please let me know if I can begin.

**Comment by Ni-zav** [comment]
I am starting to contribute to this react.dev, may I know is it okay to translate the title in the frontmatter?

**Comment by smartcoder0777** [comment]
Hi @gaearon, 
It looks like there isn't a translation in Spanish.
If you need it, I can work on it.
Looking forward to hearing from you.
Thanks,
Pavel

**Comment by tdd** [comment]
Hey @smartcoder0777.

Not sure what gave you that idea. ES has been one of the first langs to reach completion.

https://es.react.dev/ • https://github.com/reactjs/es.react.dev

Not sure why it doesn't show up on translations.react.dev anymore, though, but judging by their open PRs they're pretty up-to-date.

**Comment by smartcoder0777** [comment]
Hi @tdd , 
Thanks for your response.
By the way I'd like to take part in contributing on React.dev.
I saw that there are many issues with different tags.
Would you like to let me know which tag has a priority or good for new comers?

Best,
Pavel

**Comment by tdd** [comment]
Hey @smartcoder0777!  I'm actually not involved these days in the translations anymore… FR used to be best-of-breed and is now lagging almost a year behind.

I'm not the proper contact point when it comes to contributing to the upstream repo or some other translation, but do feel free to simply write PRs to address outstanding tickets, especially if such issues were acknowledged by the docs team.

**Comment by smartcoder0777** [comment]
Hi @tdd ,
You look like misunderstood me. I can translate documentations in several languages but I am a developer with experiences in React and Node.js.
So, I'd like to contribute to your repository but as you know, there are so many issues with many labels.
I am not sure, which one is good for me to start. So, if you can guide me once, I will appreciate it.

Thanks,
Pavel

**Comment by tdd** [comment]
Hey @smartcoder0777! Sorry about the misunderstanding.

[This link](https://github.com/reactjs/react.dev/issues?q=is%3Aissue%20state%3Aopen%20(label%3A%22good%20first%20issue%22%20OR%20label%3A%22type%3A%20typos%22)%20created%3A%3E%40today-3y) filters this repo's issues on open issues less than 3 years old (so they don't pertain to legacy docs) that are either typos or "good first issue" stuff.

That's a good springboard for contributing. Now, some of these typos may well not exist anymore (core team's docs updates rarely if ever look at issues their updates might be closing), so do double-check, but that should help sift stuff down.

I also don't think legacy docs issues should remain open, but before anyone goes on mass-closing them, perhaps getting approval from a core docs team member would be best!
